src/qrcore.cpp

```diff
@@ -3,14 +3,47 @@
 
 #define DB this->db
 
+RCoreLocked::RCoreLocked(RCore *core)
+    : core(core)
+{
+    r_th_lock_enter(core->lock);
+}
+
+RCoreLocked::RCoreLocked(RCoreLocked&& o)
+{
+    core = o.core;
```

Comment:
Not 100% sure, but shouldn't the move constructor std:.move/swap the data, so that `o`s member is invalid after the call?